src/ca65/scanner.c

```diff
@@ -769,6 +769,101 @@ static void ReadIdent (void)
     }
 }
 
+static void ReadRawStringConst (void)
+/* Called if we found the chars: 'R"'
+** Now Read a raw literal string constant into SVal.
+*/
+{
+    /* Space to store the keyword delimiter */
+    StrBuf* D = NewStrBuf ();
+
+    /* Skip the leading " string terminator */
+    NextChar ();
+
+    /* See if there is a delimiter defined after the: "
+    ** The delimiter string will follow the same rules as an identifier, but leading chars can be numeric.
+    */
+    while (IsIdChar (C)) {
+        /* Add character to the delimiter */
+        SB_AppendChar (D, C);
+        NextChar ();
+    }
+
+    /* next char after the delimiter has to be '(' */
+    if (C != '(') {
+        Error ("Error in raw string. Expected: `('");
+    }
+
+    /* skip the '(' */
+    NextChar ();
+
+    /* Read the string */
+    while (1) {
+
+        /* Maybe we are done ? */
+        if (C == ')') {
+            NextChar ();
+
+            /* If we had a opening delimiter, try a match to previous delimiter */
+            if (D->Len)
+            {
+                /* Does the first char match the first char of the stored delimiter ? */
+                if (C == D->Buf[0]) {
```

Comment:
(I hope that you see this message _before_ you start changing the code!)

I was only partly correct about StrBuf encapsulation.  They should be "opaque" when we use pointers to them.  But, when we allocate them ourselves, we must access their members directly (there are no functions for those operations).

Pointers to them make sense when we want to pass them between functions; but, you use `D` as a temporary variable, completely internal to `ReadRawStringConst()`.  Therefore, I now think that you should define `D` as a StrBuf (not a pointer to it); then, use `SB_Init()` and `SB_Done()` to start and stop it.  Your code would touch `Len` and `Buf` very directly (`D.Len` and `D.Buf`).  [And, of course, `SB_AppendChar (&D, C);`.]